File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        st, ts = {}, {}
        for i in range(len(s)):
            st[s[i]] = t[i]
            ts[t[i]] = s[i]
        
        new_s, new_t = "", ""
        for i in range(len(s)):
            new_s += ts[t[i]]
            new_t += st[s[i]]
        
        return s == new_s and t == new_t 
        

File: 0205-isomorphic-strings/0205-isomorphic-strings.go
func isIsomorphic(s string, t string) bool {
    smap := make(map[string]string)
    tmap := make(map[string]string)
    for i := 0; i < len(s); i++ {
        smap[string(s[i])] = string(t[i])
        tmap[string(t[i])] = string(s[i])
    }
    
    newS := ""
    newT := ""
    for i := 0; i < len(s); i++ {
        newS += tmap[string(t[i])]
        newT += smap[string(s[i])]
    }
    
    return s == newS && t == newT
}